db/versions/0009.yml

```diff
@@ -0,0 +1,288 @@
+            jsonb_build_object(
+              'PartitionKey', encode_composite_key(cache_purges.provisioner_id, cache_purges.worker_type),
+              'RowKey', cache_name,
+              'provisionerId', provisioner_id,
+              'workerType', worker_type,
+              'cacheName', cache_name,
+              'before', before,
+              'expires', expires
+            ) as value,
+            1 as version,
+            cache_purges.etag as etag from cache_purges
+          where
+            (cache_purges_entities_scan.pk is null or cache_purges_entities_scan.pk = decoded_composite_key[1] || '~' || decoded_composite_key[2]) and
+            (cache_purges_entities_scan.rk is null or cache_purges_entities_scan.rk = cache_name) and
+            case
+              when exp_cond_operator = '=' then expires = exp_cond_operand
+              when exp_cond_operator = '<' then expires < exp_cond_operand
+              when exp_cond_operator = '<=' then expires <= exp_cond_operand
+              when exp_cond_operator = '>' then expires > exp_cond_operand
+              when exp_cond_operator = '>=' then expires >= exp_cond_operand
+              else expires <> exp_cond_operand
+            end
+          order by cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name
+          limit case
+            when (size is not null and size > 0) then size + 1
+            else null
+          end
+          offset case
+            when (page is not null and page > 0) then page
+            else 0
+          end;
+        else
+          return query select
+            encode_composite_key(cache_purges.provisioner_id, cache_purges.worker_type) as partition_key,
+            encode_string_key(cache_purges.cache_name) as row_key,
+            jsonb_build_object(
+              'PartitionKey', encode_composite_key(cache_purges.provisioner_id, cache_purges.worker_type),
+              'RowKey', cache_name,
+              'provisionerId', provisioner_id,
+              'workerType', worker_type,
+              'cacheName', cache_name,
+              'before', before,
+              'expires', expires
+            ) as value,
+            1 as version,
+            cache_purges.etag as etag from cache_purges
+          where
+            (cache_purges_entities_scan.pk is null or (cache_purges.provisioner_id = decoded_composite_key[1] and cache_purges.worker_type = decoded_composite_key[2])) and
+            (cache_purges_entities_scan.rk is null or cache_purges_entities_scan.rk = cache_name)
+          order by cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name
+          limit case
+            when (size is not null and size > 0) then size + 1
+            else null
+          end
+          offset case
+            when (size is not null and size > 0 and page is not null and page > 0) then page
+            else 0
+          end;
+        end if;
+      end
+  purge_requests:
```

Comment:
How are we deciding which queries to paginate and which not to? I admit that >1000 cache purges for a workertype seems unlikely but this is a more general question

Reply:
I think the easy answer is "where the REST API paginates".  There may be circumstances where that's different, but as an 80% answer it's probably good.  In this case, [`purgeCache.purgeRequests`](https://docs.taskcluster.net/docs/reference/core/purge-cache/api#purgeRequests) is not paginated